src/main/java/com/thealgorithms/sorts/SmoothSort.java

```diff
@@ -0,0 +1,30 @@
+package com.thealgorithms.sorts;
+
+import com.thealgorithms.datastructures.heaps.LeonardoHeap;
+
+/**
+ * Wikipedia: https://en.wikipedia.org/wiki/Smoothsort
+ */
+public final class SmoothSort implements SortAlgorithm {
+
+    public SmoothSort() {
```

Comment:
You don't need to write an empty contstructor; the Java compiler will automatically insert one for you (only if you haven't defined any other constructors that take some args).